src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/ResourceUtilizationCounters.cs

```diff
@@ -7,24 +7,24 @@ namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring;
 /// Represents the names of instruments published by this package.
 /// </summary>
 /// <remarks>
-/// These counters are currently only published on Linux.
+/// These metrics are currently only published on Linux.
 /// </remarks>
 /// <seealso cref="System.Diagnostics.Metrics.Instrument"/>
 public static class ResourceUtilizationCounters
 {
     /// <summary>
-    /// Gets the CPU consumption of the running application in percentages.
+    /// Gets the CPU consumption of the running application in range <c>[0, 1]</c>.
     /// </summary>
     /// <remarks>
     /// The type of an instrument is <see cref="System.Diagnostics.Metrics.ObservableGauge{T}"/>.
     /// </remarks>
-    public static string CpuConsumptionPercentage => "cpu_consumption_percentage";
+    public static string CpuUtilization => "process.cpu.utilization";
 
     /// <summary>
-    /// Gets the memory consumption of the running application in percentages.
+    /// Gets the memory consumption of the running application in range <c>[0, 1]</c>.
     /// </summary>
     /// <remarks>
     /// The type of an instrument is <see cref="System.Diagnostics.Metrics.ObservableGauge{T}"/>.
     /// </remarks>
-    public static string MemoryConsumptionPercentage => "memory_consumption_percentage";
+    public static string MemoryUtilization => "process.memory.utilization";
```

Comment:
Good point, I was following https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/process-metrics.md, but there's no such thing we have here. Will add `virtual` then since it's the right "type" of memory we measure here.

Reply:
From a little web searching, usage_in_bytes appears to be RSS + swap cache + swap, which approximately corresponds to committed virtual memory. Looking on Windows the container case appears aligned with that (https://github.com/dotnet/extensions/blob/2967ec96d4d3a46cf43e54f8ff5ef22e4e5d69e1/src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsContainerSnapshotProvider.cs#L151), but the non-container case is very different:
https://github.com/dotnet/extensions/blob/2967ec96d4d3a46cf43e54f8ff5ef22e4e5d69e1/src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsSnapshotProvider.cs#L36

I think we need all cases to agree on the same memory abstraction or it will be very confusing trying to name this metric or describe it to users. I don't think of this as a naming problem, its an inconsistency in the underlying abstraction. I'd suggest we either get it resolved or avoid making this memory metric public.

Reply:
I see. Renamed the instrument to be `process.memory.virtual.utilization`.
Regarding Windows - we don't emit anything via `Meter` or `IMeterFactory` there yet, thus it will be only pushed via `IResourceUtilizationPublisher` defined by a user. But I agree to your point, now it's `VirtualMemorySize64` for non-container case. Does it seem correct now?

Reply:
No unfortunately VirtualMemorySize64 measures reserved virtual memory. That number can be (and usually is) vastly different than committed virtual memory (OTel's instrumentation appears to have the same issue and we missed it!). Pretty sure you'll need to pinvoke to the OS to get this value.

Here is an example app you can test with:
```C#
using System.Diagnostics;
using System.Runtime.InteropServices;

// You can experiment with manually allocating reserved or committed memory
//uint reserveSize = 2_000_000_000;
//uint commitSize = 100_000_000;
//IntPtr reservedMem = VirtualAlloc(IntPtr.Zero, reserveSize, AllocationType.Reserve, MemoryProtection.ReadWrite);
//IntPtr commitMem = VirtualAlloc(IntPtr.Zero, commitSize, AllocationType.Commit, MemoryProtection.ReadWrite);
//Console.WriteLine($"Reserved {reserveSize/1_000_000_000.0} GB of memory");
//Console.WriteLine($"Committed {commitSize / 1_000_000_000.0} GB of memory");

Console.WriteLine($"Process.VirtualMemorySize64= {Process.GetCurrentProcess().VirtualMemorySize64 / 1_000_000_000.0} GB");
Console.ReadLine();


[DllImport("kernel32.dll", SetLastError = true, ExactSpelling = true)]
static extern IntPtr VirtualAlloc(IntPtr lpAddress,
   uint dwSize, AllocationType flAllocationType, MemoryProtection flProtect);

[Flags]
public enum AllocationType : uint
{
    Commit = 0x1000,
    Reserve = 0x2000,
}

[Flags]
public enum MemoryProtection : uint
{
    ReadWrite = 0x04,
}
```

When I run this locally it reports:
```
Process.VirtualMemorySize64= 2204.04000768 GB
```

Also I was looking more at the code and I see in some cases it is using a commit limit and in other cases it is using physical memory as the denominator of the ratio. These mismatched measurement abstractions seem weird and ratios larger than 1 would certainly be possible, maybe even desirable/healthy. I think reporting raw numbers and naming them accurately rather than inventing novel ratios would make this a lot more comprehensible.

Reply:
I created #4538 to track that. The behavior doesn't affect metrics (as I mentioned we don't push Windows measurements to metrics yet - #4477) and I don't want to extend scope of this PR more. Feel free to unresolve the discussion.